packages/astro/src/compiler/transform/postcss-scoped-styles/index.ts

```diff
@@ -96,10 +96,14 @@ export function scopeRule(selector: string, className: string) {
 
 /** PostCSS Scope plugin */
 export default function astroScopedStyles(options: AstroScopedOptions): Plugin {
+  const rulesScopedCache = new WeakSet();
```

Comment:
Nice!

Reply:
Actually, are we sure that this isn’t too aggressive? E.g. would this cache get hit if someone has a `.button` in separate files that should get different scopes?